pkg/signal/main.go

```diff
@@ -0,0 +1,70 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package signal
+
+import (
+	"sync"
+
+	"github.com/zeebo/errs"
+
+	"storj.io/storj/pkg/utils"
+)
+
+// ErrNilCallback throws if there is no callback function provided
+var ErrNilCallback = errs.Class("must provide a callback function to register")
+
+// Dispatcher is what's loaded onto each service
+type Dispatcher struct {
+	sync.RWMutex
```

Comment:
It doesn't look like you need an RWMutex over a Mutex. If all you need is a Mutex, it would be more efficient to just use it.